script.js

```diff
@@ -1,4 +1,602 @@
+  if (diceNumberChosen === 1) {
+    chosenNumber = getCombinedNumber(diceList[0], diceList[1]);
+  } else {
+    chosenNumber = getCombinedNumber(diceList[1], diceList[0]);
+  }
+
+  // Store player num in the relevant global player num variable
+  if (currentPlayer === 1) {
+    player1CombinedNumber = chosenNumber;
+  } else {
+    player2CombinedNumber = chosenNumber;
+  }
+
+  // Return generated player num to parent function
+  return chosenNumber;
+};
+
+// Check who wins
+var checkWinner = function () {
+  if (player1CombinedNumber > player2CombinedNumber) {
+    return 1;
+  }
+  return 2;
+};
+
+// More comfortable - lowest combined number
+// var generatePlayerNumber = function () {
+//   // Check current list and assign to access accordingly
+//   var diceList = null;
+//   var chosenNumber = null;
+
+//   if (currentPlayer === 1) {
+//     diceList = player1DiceRolls;
+//   } else {
+//     diceList = player2DiceRolls;
+//   }
+
+//   sortList(diceList);
+
+//   // Sort list small to big
+//   // Lowest Combined Number Mode
+//   // Add a game mode such that the player with the lowest combined number is the winner.
+//   while (diceList.length > 0) {
+//     var biggestNumber = diceList.pop();
+
+//     // 1st iteration is just number itself
+//     if ((chosenNumber = null)) {
+//       chosenNumber = biggestNumber;
+//     } // Combine to get number for lowest
+//     else {
+//       if (inputGameMode == LOWEST_COMBINED_NUMBER_GAME_MODE) {
+//         chosenNumber = getCombinedNumber(biggestNumber, chosenNumber);
+//       } // If not it's normal mode, biggest number
+//       else {
+//         chosenNumber = getCombinedNumber(chosenNumber, biggestNumber);
+//       }
+//     }
+//   }
+
+//   // Store player num in the relevant global player num variable
+//   if (currentPlayer === 1) {
+//     player1CombinedNumber = chosenNumber;
+//   } else {
+//     player2CombinedNumber = chosenNumber;
+//   }
+
+//   // Return generated player num to parent function
+//   return chosenNumber;
+// };
+
+// More comfortable - auto generation for variable modes
+var generatePlayerNumber = function () {
+  // Check current list and assign to access accordingly
+  var diceList = playerInfo[currentPlayer].diceRolls;
+  var chosenNumber;
+
+  sortList(diceList);
+
+  // Sort list small to big
+  // Lowest Combined Number Mode
+  // Add a game mode such that the player with the lowest combined number is the winner.
+  while (diceList.length > 0) {
+    var biggestNumber = diceList.pop();
+
+    // 1st iteration is just number itself
+    if ((chosenNumber = null)) {
+      chosenNumber = biggestNumber;
+    } // Combine to get number for lowest
+    else {
+      if (inputGameMode == LOWEST_COMBINED_NUMBER_GAME_MODE) {
+        chosenNumber = getCombinedNumber(biggestNumber, chosenNumber);
+      } // If not it's normal mode, biggest number
+      else {
+        chosenNumber = getCombinedNumber(chosenNumber, biggestNumber);
+      }
```

Comment:
This will always return `chosenNumber` as `NaN` because the `chosenNumber` that you give as input into the `getCombinedNumber` function is always `undefined`.